### ISSUE TITLE
$JSObjectDict.__setattr__ for functions, wrongly assumes exceptions always have arguments

### ISSUE THREAD

**Issue by nirhershko** [brython-dev/brython#539]
repr(undefined) fails when i do `raise NotImplementedError()`.

```
TypeError: Cannot read property '__class__' of undefined
    at Object.repr (..libs/Brython/brython.js:5662:26)
    at Error.err.toString (..libs/Brython/brython.js:6792:10)
    at self.js.(anonymous function) (..libs/Brython/brython.js:6793:16)
```

brython.js is the minified 3.2.9:
```
5662: function repr(obj){if(obj.__class__===$B.$factory){

6788: try{return value.apply(null,args)}
6789: catch(err){err=$B.exception(err)
6790: var info=_b_.getattr(err,'info')
6791: err.toString=function(){return info+'\n'+err.__class__.__name__+
6792: ': '+_b_.repr(err.args[0])}
6793: console.log(err+'')
```

The code `NotImplementedError()`, for some reason, is interpreted as `NotImplementedError(undefined)`.

**I could not reproduce it easily** - just `raise NotImplementedError()` works as expected. but for some reason in my program (which includes a call to a python function from javascript from angular) this error happens, and because of this error, I don't see my python stack track (so I can't see what caused my exception).
[my workaround below fixes it for me]

It may be worth looking into why `raise NotImplementedError()` sets undefined as a parameter sometimes,
but more importantly, I think `repr(undefined)` should not crash.

I added the following in line 5662:
```
function repr(obj){
  if (obj === undefined) {
    return "undefined"
  }
  if(obj.__class__===$B.$factory){
  ...
```

and it worked - I got the python stacktrace (in addition to the js stacktrace that followed):
```
Traceback (most recent call last):
  ....
  module __main__ line 133
    raise NotImplementedError()
NotImplementedError: undefined
```

What do you think about this solution?

**Comment by nirhershko** [comment]
Hmm..
It seems that the problem is actually in the code that handle python function calls from JS - 
it assumes that every exception thrown will have at least one argument (by always accessing err.args[0])

https://github.com/brython-dev/brython/blob/97f8a0562bb6e19a4fc04f651fcd65db07739c73/www/src/js_objects.js#L391

**Comment by nirhershko** [comment]
should be very easy to fix - but I'm not sure - is there any existing common code for printing exceptions?

**Comment by nirhershko** [comment]
To reproduce:
```
<script type="text/python3">
def foo():
  raise NotImplementedError()
from browser import window
window.foo = foo
</script>
```
```
<body>
<input type="button" onclick="foo()" value="Foo">
...
```

Expected:
```
Traceback (most recent call last):
  module __main__ line 3
    raise NotImplementedError()
NotImplementedError
```

**Comment by nirhershko** [comment]
re:
> perhaps exceptions should always have at least one argument

(in https://github.com/jonathanverner/brython/commit/34dea104ff682fec2e9525dc5b658a6cce351aad) they don't.

try in CPython:
```
>>> raise NotImplementedError()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NotImplementedError
>>> raise NotImplementedError("foo")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NotImplementedError: foo
>>> raise NotImplementedError("foo", 0xba)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NotImplementedError: ('foo', 186)
```

**Comment by PierreQuentel** [comment]
I think this was fixed in Jonathan's PR that I just merged. Can you confirm and if so close the issue ?

Thanks,
Pierre

**Comment by nirhershko** [comment]
I think it's too bad that there isn't one place that prints exceptions.
IIUC, the fix still doesn't print more than one argument, like `traceback.print_exc()`.

---

There's another exception-printing related problem that I might as well keep on the same bug

To reproduce:
```
<script type="text/python3">
import file1
</script>
```

file1.py:
```
import file2
```

file2.py:
```
random.random()
```

What I expect to see in the console is `NameError: name 'random' is not defined` **and the traceback**

I found two issues here, though I might be off:
1. https://github.com/brython-dev/brython/blob/22f1e231a78ddbcd421549cfd2e2f32feb9d074b/www/src/py2js.js#L7502 doesn't print the stack. I'm not sure how to set `onerror`, but I guess it should print the exception info on the console anyway..
2. when adding `console.log(_b_.getattr(e,'info'));` there, I still didn't get the stack (it only had: `Traceback (most recent call last):` and that's it).
I traced it down to
https://github.com/brython-dev/brython/blob/1e0cb7d878d6542ddc22415a6e694d7f0de9d114/www/src/py_exceptions.js#L222 which looks at `frame[1].$line_info` though in practice i found it in `frame[3].$line_info`

fixing both allowed me to see the stack trace (but again, I don't know the code so I'm not sure what frame[1] should contain, and also I didn't run any of tests).
